### PR TITLE
fix iOS memory leaks while app have PlatformView

### PR DESCRIPTION
**Pull request by linyu92**

### CODE REVIEW

File: shell/platform/darwin/ios/framework/Source/FlutterPlatformViews.mm [hunk 1]
```diff
@@ -464,7 +464,7 @@
       return;
     }
     fml::scoped_nsobject<FlutterOverlayView> overlay_view(
-        [[[FlutterOverlayView alloc] init] retain]);
+        [[FlutterOverlayView alloc] init]);
```
Comment (dnfield) [inline shell/platform/darwin/ios/framework/Source/FlutterPlatformViews.mm]: this is not safe - scoped_nsobject does not automatically retain the object and you could end up with a dead pointer here.

File: shell/platform/darwin/ios/framework/Source/FlutterPlatformViews.mm [hunk 2]
```diff
@@ -489,7 +489,7 @@
   }
   auto contentsScale = flutter_view_.get().layer.contentsScale;
   fml::scoped_nsobject<FlutterOverlayView> overlay_view(
-      [[[FlutterOverlayView alloc] initWithContentsScale:contentsScale] retain]);
+      [[FlutterOverlayView alloc] initWithContentsScale:contentsScale]);
```
Comment (dnfield) [inline shell/platform/darwin/ios/framework/Source/FlutterPlatformViews.mm]: ditto.

**Comment by cyanglaz** [comment]
This fix is already in master. What was the reason of patching an old hotfix branch?
Anyways, changes look good to me after fixing @dnfield's suggestions.

**Comment by dnfield** [comment]
Ah.  I didn't even notice the branch.  This should go through a hotfix request first.

**Comment by chinmaygarde** [comment]
Closing as obsolete.

**Pull request closed without merging**